### PR TITLE
[Feat] Review Search - Integration

### PR DESCRIPTION
**Pull request by jessiiepeng**
## Notion ticket link

[Review Search - Integration](https://www.notion.so/uwblueprintexecs/16fb020b93f641c8b384f76a1b3c769e?v=aedc8d99cd8b437cb556ece10f3c0e3c&p=26e4a32aefc146389df47736fbd5c856&pm=c)

## Implementation description
* added additional parameters `search`, `size` and `page` to `getReviews` function in ReviewAPIClient.
* calls search endpoint to fetch reviews when updated query
* sort review books so thumbnail always displays first book in series

## Steps to test
1. Go to http://localhost:3000/magazine/search_results/ and verify that books are displayed
2. Search up by author name, book name, etc and verify that books are displayed
3. Verify that first book in series for review is shown in thumbnail

## What should reviewers focus on?
* 

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have run docker-compose up and my project compiled
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/components/pages/SearchReviews/SearchReviews.tsx [hunk 1]
```diff
@@ -78,9 +79,13 @@ const SearchReviews = (): React.ReactElement => {
...
-    setDisplayedReviews(mockReviews);
-
-    setLoading(false);
+    reviewAPIClient
+      .getReviews(searchText, 25, 0)
+      .then((reviewResponse: PaginatedReviewResponse) => {
+        setDisplayedReviews(mapReviewResponseToReview(reviewResponse.reviews));
+        setLoading(false);
+      });
+    // setDisplayedReviews(mockReviews); // todo remove this line
```
Comment (williamtran10) [inline frontend/src/components/pages/SearchReviews/SearchReviews.tsx:88]: is this supposed to be removed?

File: frontend/src/components/pages/SearchReviews/SearchReviews.tsx [hunk 2]
```diff
@@ -78,27 +79,33 @@ const SearchReviews = (): React.ReactElement => {
...
+    reviewAPIClient
+      .getReviews(searchText, 25, 0)
+      .then((reviewResponse: PaginatedReviewResponse) => {
+        setDisplayedReviews(mapReviewResponseToReview(reviewResponse.reviews));
+        setLoading(false);
+      });
   }, [searchText, genresFilter, ageRangeFilter]);
 
   if (loading) {
     return <LoadingSpinner mt="21%" />;
```
Comment (susheelk) [inline frontend/src/components/pages/SearchReviews/SearchReviews.tsx:91]: I think it looks better if you only cover the grid with the loading spinner, not the whole page (ie leave the search bar)

File: frontend/src/components/pages/SearchReviews/SearchReviews.tsx [hunk 3]
```diff
@@ -78,27 +79,33 @@ const SearchReviews = (): React.ReactElement => {
...
   }
   return (
```
Comment (susheelk) [inline frontend/src/components/pages/SearchReviews/SearchReviews.tsx:93]: I think it looks better if you only cover the grid with the loading spinner, not the whole page (ie leave the search bar)

Comment (jessiiepeng) [reply frontend/src/components/pages/SearchReviews/SearchReviews.tsx:90]: I agree. Pushed up the changes!

**Review by susheelk — APPROVED** [review]
LGTM, Nice work! :slightly_smiling_face:

**Pull request merged**